lib/models/introductory_price.dart

```diff
@@ -19,6 +15,8 @@ enum PeriodUnit {
 
 /// Contains all the introductory information associated with a [Product]
 class IntroductoryPrice with _$IntroductoryPrice {
+  const IntroductoryPrice._();
```

Comment:
needed to add this to enable the custom getter of the PeriodUnit type. see https://github.com/rrousselGit/freezed#custom-getters-and-methods